tests/Integration/inc/Engine/CriticalPath/RESTGenerate/generate.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace WP_Rocket\Tests\Integration\inc\Engine\CriticalPath\RESTGenerate;
+
+use Brain\Monkey\Functions;
+use WP_Site;
+use WP_Rocket\Tests\Integration\RESTVfsTestCase;
+
+/**
+ * @covers \WP_Rocket\Engine\CriticalPath\RESTGenerate::delete
+ * @group  CriticalPath
+ * @group  vfs
+ */
+class Test_Generate extends RESTVfsTestCase {
+	protected $path_to_test_data = '/inc/Engine/CriticalPath/RESTGenerate/generate.php';
+	private static $post_id;
+
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$post_id = $factory->post->create();
+	}
+
+	protected function doTest( $site_id, $config, $expected ) {
+		if ( isset( $config['post_data'] ) ) {
+			$config['post_data']['ID'] = self::$post_id;
+
+			$post_id = wp_update_post( $config['post_data'], true );
+		} else {
+			$post_id = 0;
+		}
```

Comment:
Why not create the post here if there's `'post_data'`, i.e. instead of in `wpSetUpBeforeClass`? 

```suggestion
		if ( isset( $config['post_data'] ) ) {
			$config['post_data']['ID'] = $post_id = $factory->post->create( $config['post_data'] );
		} else {
			$post_id = 0;
		}
```

Reply:
I'm creating the post in `wpSetupBeforeClass` to only create one that is updated for each test, instead of creating a new one each time. Else in the test the post ID is incrementing, and it makes the expectations less reliable.